frontend/src/__mocks__/mockInferenceServiceK8sResource.ts

```diff
@@ -13,7 +13,26 @@ type MockResourceConfigType = {
   url?: string;
 };
 
-export const mockInferenceServicek8sError = () => ({
+type InferenceServicek8sError = {
+  kind: string;
+  apiVersion: string;
+  metadata: Record<string, unknown>;
+  status: string;
+  message: string;
+  reason: string;
+  details: {
+    name: string;
+    group: string;
+    kind: string;
+    causes: {
+      reason: string;
+      message: string;
+      field: string;
+    }[];
+  };
+};
+
+export const mockInferenceServicek8sError = (): InferenceServicek8sError => ({
```

Comment:
we do have k8sStatus in [k8types.ts](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/k8sTypes.ts#L232-L239) file in frontend folder ,Can we make use of that and write it in this way in src/types.ts?

```
export type InferenceStatus = K8sStatus & {
  metadata: Record<string, unknown>;
  details: {
    name: string;
    group: string;
    kind: string;
    causes: {
      reason: string;
      message: string;
      field: string;
    }[];
  };
};

```
Is it okay if I change the code key to optional in K8sTypes.ts? since we don't have it here

Reply:
@pnaik1 I'd like to match the kubernetes Status definition. I just need to find it first :) I'll report back

Reply:
Let's address this in a followup PR because having to rebase this issue to keep up with changes is painful.